string-manipulation-3/src/title-case.js

```diff
@@ -11,81 +11,83 @@
-        break;
-      case 'javascript:':
-        titleFinal.push('JavaScript:');
-        break;
-
-      default:
-        if (titleSplit[i].includes('-') === true) {
-          var hyphSplit = titleSplit[i].split('-');
-          var hyphCapped = hyphSplit[0][0].toUpperCase() + hyphSplit[0].slice(1, hyphSplit[0].length) + '-' + hyphSplit[1][0].toUpperCase() + hyphSplit[1].slice(1, hyphSplit[1].length);
-          titleFinal.push(hyphCapped);
-        } else {
-          titleFinal.push(titleSplit[i][0].toUpperCase() + titleSplit[i].slice(1, titleSplit[i].length));
-        }
+    if (titleSplit[i].toLowerCase() === 'javascript') {
+      titleFinal.push('JavaScript');
+
+    } else if (titleSplit[i].toLowerCase() === 'javascript:') {
+      titleFinal.push('JavaScript:');
+
+    } else if (i === 0) {
+      titleFinal.push(titleSplit[i][0].toUpperCase() + titleSplit[i].slice(1, titleSplit[i].length));
+
+    } else if (i > 0 && titleSplit[i - 1].includes(':')) {
+
+      titleFinal.push(titleSplit[i][0].toUpperCase() + titleSplit[i].slice(1, titleSplit[i].length));
+
+    } else if (titleSplit[i].includes('-') === true) {
+      var hyphSplit = titleSplit[i].split('-');
+      var hyphCapped = hyphSplit[0][0].toUpperCase() + hyphSplit[0].slice(1, hyphSplit[0].length) + '-' + hyphSplit[1][0].toUpperCase() + hyphSplit[1].slice(1, hyphSplit[1].length);
+      titleFinal.push(hyphCapped);
+    } else if (titleSplit[i].length > 3) {
+      titleFinal.push(titleSplit[i][0].toUpperCase() + titleSplit[i].slice(1, titleSplit[i].length));
+
+    } else if (titleSplit[i] === 'api') {
+      titleFinal.push('API');
+    } else if (titleSplit[i] === 'and') {
+      titleFinal.push(titleSplit[i]);
+
+    } else if (titleSplit[i] === 'or') {
+      titleFinal.push(titleSplit[i]);
+
+    } else if (titleSplit[i] === 'nor') {
+      titleFinal.push(titleSplit[i]);
+
+    } else if (titleSplit[i] === 'but') {
+      titleFinal.push(titleSplit[i]);
+
+    } else if (titleSplit[i] === 'a') {
+      titleFinal.push(titleSplit[i]);
+
+    } else if (titleSplit[i] === 'as') {
+      titleFinal.push(titleSplit[i]);
+
+    } else if (titleSplit[i] === 'at') {
+      titleFinal.push(titleSplit[i]);
+
+    } else if (titleSplit[i] === 'by') {
+      titleFinal.push(titleSplit[i]);
+
+    } else if (titleSplit[i] === 'for') {
+      titleFinal.push(titleSplit[i]);
+
+    } else if (titleSplit[i] === 'in') {
+      titleFinal.push(titleSplit[i]);
+
+    } else if (titleSplit[i] === 'of') {
+      titleFinal.push(titleSplit[i]);
+
+    } else if (titleSplit[i] === 'on') {
+      titleFinal.push(titleSplit[i]);
+
+    } else if (titleSplit[i] === 'per') {
+      titleFinal.push(titleSplit[i]);
+
+    } else if (titleSplit[i] === 'to') {
+      titleFinal.push(titleSplit[i]);
+    } else if (titleSplit[i] === 'the') {
+      titleFinal.push(titleSplit[i]);
+
+    } else if (titleSplit[i] === 'an') {
+      titleFinal.push(titleSplit[i]);
```

Comment:
Rather than have all these conditional blocks, I would make an array of all the exception words and check if the current word is present in that array using the `indexOf` method:

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/indexOf